### ISSUE TITLE
Ensure CI builds for branches in 3rd party repositories build successfully 

### ISSUE THREAD

**Issue by lucaswoj** [mapbox/mapbox-gl-js#2465]
ref https://github.com/mapbox/mapbox-gl-js/pull/2453

**Comment by mourner** [comment]
I know remember that there was a known limitation to PR branches on Circle, but @jfirebaugh should know more.

**Comment by lucaswoj** [comment]
This looks like Coverall's fault. If there isn't an easy solution, I think we should just ignore errors from the coveralls CLI.

**Comment by jfirebaugh** [comment]
https://github.com/lemurheavy/coveralls-public/issues/632 suggests some troubleshooting steps.

**Comment by lucaswoj** [comment]
Looks like we need to set `COVERALLS_REPO_TOKEN` in our CI environment. I have mitigated the problem temporarily by preventing a coveralls error from causing the build to fail.

**Comment by lucaswoj** [comment]
I researched this a little further and found that CircleCI intentionally omits environment variables from forked builds. It even says so in the build log:

<img width="967" alt="screen shot 2016-12-05 at 3 33 53 pm" src="https://cloud.githubusercontent.com/assets/281306/20907017/5351748e-bb00-11e6-9fe9-77b3c4fabcd8.png">

The only way we could enable coveralls for forked builds is to make our coveralls repo token public.

As a compromise, I have removed "coveralls" from the list of required checks.

Long term: I wonder if we can hook the coverage stats into our benchmarking system #1362 to get around this design flaw in coveralls.
